api/controllers/UserController.js

```diff
@@ -214,26 +215,168 @@ module.exports = {
+          flair2.flair_text.match(/(\d{4}-){2}\d{4}/g),
+          logged_fcs,
+          req.params.additionalFCs
+        );
+        var igns = flair1.flair_text.substring(flair1.flair_text.indexOf("||") + 3);
+        var ptradesBanPromise = new Promise(function(resolve, reject) {
+          Ban.banFromSub(req.user.redToken, req.params.username, req.params.banMessage, req.params.banNote, 'pokemontrades', duration, resolve, reject);
+        });
+        var svexBanPromise = new Promise(function(resolve, reject) {
+          Ban.banFromSub(req.user.redToken, req.params.username, req.params.banMessage, req.params.banNote, 'SVExchange', duration, resolve, reject);
+        });
+        var promises;
+        if (duration) {
+          promises = [ //Tasks for tempbanning
+            ptradesBanPromise, 
+            svexBanPromise
+          ];
+        } else {
+          var ptradesFlairPromise = new Promise(function(resolve, reject) {
+            Ban.giveBannedUserFlair(req.user.redToken, req.params.username, flair1.flair_css_class, flair1.flair_text, 'pokemontrades', resolve, reject);
+          });
+          var svexFlairPromise = new Promise(function(resolve, reject) {
+            Ban.giveBannedUserFlair(req.user.redToken, req.params.username, flair2.flair_css_class, flair2.flair_text, 'SVExchange', resolve, reject);
+          });
+          var ptradesAutomodPromise = new Promise(function(resolve, reject) {
+            Ban.updateAutomod(req.user.redToken, req.params.username, 'pokemontrades', unique_fcs, resolve, reject);
+          });
+          var svexAutomodPromise = new Promise(function(resolve, reject) {
+            Ban.updateAutomod(req.user.redToken, req.params.username, 'SVExchange', unique_fcs, resolve, reject);
+          });
+          var removeTSVPromise = new Promise(function(resolve, reject) {
+            Ban.removeTSVThreads(req.user.redToken, req.params.username, resolve, reject);
+          });
+          var updateBanlistPromise = new Promise(function(resolve, reject) {
+            Ban.updateBanlist(req.user.redToken, req.params.username, req.params.banlistEntry, unique_fcs, igns, resolve, reject);
+          });
+          var localBanPromise = new Promise(function(resolve, reject) {
+            Ban.localBanUser(req.params.username, resolve, reject);
+          });
+          promises = [ //Tasks for permabanning
+            ptradesBanPromise,
+            svexBanPromise,
+            ptradesFlairPromise,
+            svexFlairPromise,
+            ptradesAutomodPromise,
+            svexAutomodPromise,
+            removeTSVPromise,
+            updateBanlistPromise,
+            localBanPromise
+          ];
+        }
+        Promise.all(promises).then(function(result) {
+          res.json('ok', 200);
+        }, function(error) {
+          console.log(error);
+          res.json({error: error}, 500);
+        });
+      }, req.params.username);
     });
   },
 
```

Comment:
I decided to keep the backend for this even though we removed the button for locally banning/unbanning people. We shouldn't have to use this too often, but it could come in handy if we unban someone after an appeal. (Without this, there's no easy way to remove a user from the local banlist.)